client/src/components/layout/MainLayout.js

```diff
@@ -63,152 +63,134 @@ const Button = styled.button`
       };
-
+  
       const savedMessage = await parseService.addMessage(
-        selfChatId,
-        activeChat.id,
+        selfId, 
+        activeFriend.id,
         newMessage
       );
-      setMessages((prevMessages) => [...prevMessages, savedMessage]);
-      setInputValue(""); // Clear input after sending
+  
+      setMessages(prevMessages => [...prevMessages, savedMessage]);
+      setInputValue(''); // Clear input after sending
     } catch (error) {
-      console.error("Failed to send message:", error);
+      console.error('Failed to send message:', error);
     }
   };
   return (
     <Container>
       <Sidebar>
-        <NewRecipient
-          placeholder="New recipient"
-          onChatAdded={handleChatAdded}
+        <NewRecipient 
+          placeholder="New recipient" 
+          onConnect={onConnect}
+          selfId={selfId}
         />
-
-        <ChatList>
+        
+        <RecipientList>
           {loading ? (
-            <div>Loading chats...</div>
+            <div>Loading friends...</div>
           ) : error ? (
             <div>{error}</div>
           ) : (
-            chats.map((chat) => (
-              <ChatItem
-                key={chat.id}
-                id={chat.id}
-                name={chat.name}
-                message={chat.message}
-                isPinned={chat.isPinned}
-                active={chat.active}
-                avatar={chat.avatar}
-                onClick={() => handleChatClick(chat.id)}
-                onPinUpdate={handlePinUpdate}
+            friends.map(friend => (
+              <RecipientItem
+                key={friend.id}
+                id={friend.id}
+                name={friend.name}
+                avatar={friend.avatar}
+                active={friend.active}
+                onClick={() => handleRecipientClick(friend.id)}
               />
             ))
           )}
-        </ChatList>
+        </RecipientList>
       </Sidebar>
 
       <ChatArea>
-        <MessagesContainer>
-          <Message messages={messages} />
-        </MessagesContainer>
-
-        <MessageInput>
-          <Input
-            value={inputValue}
-            onChange={(e) => setInputValue(e.target.value)}
-            placeholder="Type a message..."
-            onKeyUp={(e) => {
-              if (e.key === "Enter") {
-                handleSendMessage();
-              }
-            }}
-          />
-          <Button onClick={handleSendMessage}>Send</Button>
-        </MessageInput>
+
+      <MessagesContainer>
+        <Message messages={messages} />
+       </MessagesContainer>
+
+       <MessageInput>
+      <Input 
+        value={inputValue}
+        onChange={(e) => setInputValue(e.target.value)}
+        placeholder="Type a message..."
+        onKeyPress={e => {
+          if (e.key === 'Enter') {
+            handleSendMessage();
+          }
+        }}
+      />
+      <Button onClick={handleSendMessage}>Send</Button>
+    </MessageInput>
```

Comment:
> Use a styled `form` element instead of a styled `div` as a wrapper to the `<Input/>` element. This will ensure proper usability and accessibility, such as keyboard focusability and submission when hitting "Enter".

I prefer input + button instead of form in this case. Some default behaviors of form is not expected and cause bugs (in todo list, we need event.preventDefault()). Of course in this case form is fine.